OpenRA.Mods.Common/UtilityCommands/CheckYaml.cs

```diff
@@ -77,10 +77,25 @@ void IUtilityCommand.Run(Utility utility, string[] args)
 						}
 					}
 
-					modData.MapCache.LoadMaps();
-					maps.AddRange(modData.MapCache
-						.Where(m => m.Status == MapStatus.Available)
-						.Select(m => new Map(modData, m.Package)));
+					foreach (var kv in modData.Manifest.MapFolders)
+					{
+						var name = kv.Key;
+						var classification = string.IsNullOrEmpty(kv.Value)
+							? MapClassification.Unknown : Enum<MapClassification>.Parse(kv.Value);
+
+						if (classification == MapClassification.Unknown)
+							continue;
+
+						var optional = name.StartsWith("~", StringComparison.Ordinal);
+						if (optional)
+							name = name.Substring(1);
+
+						using (var package = (IReadWritePackage)modData.ModFiles.OpenPackage(name))
+						{
+							foreach (var map in package.Contents)
```

Comment:
This needs to filter out things that aren't packages.  It currently crashes for me because `.DS_Store` is not a valid map package and triggers an NRE.

The following works for me:
```diff
diff --git a/OpenRA.Mods.Common/UtilityCommands/CheckYaml.cs b/OpenRA.Mods.Common/UtilityCommands/CheckYaml.cs
index ce96f7cb2f..4c1df7cef9 100644
--- a/OpenRA.Mods.Common/UtilityCommands/CheckYaml.cs
+++ b/OpenRA.Mods.Common/UtilityCommands/CheckYaml.cs
@@ -93,7 +93,11 @@ void IUtilityCommand.Run(Utility utility, string[] args)
                                                using (var package = (IReadWritePackage)modData.ModFiles.OpenPackage(name))
                                                {
                                                        foreach (var map in package.Contents)
-                                                               maps.Add(new Map(modData, package.OpenPackage(map, modData.ModFiles)));
+                                                       {
+                                                               var mapPackage = package.OpenPackage(map, modData.ModFiles);
+                                                               if (mapPackage != null)
+                                                                       maps.Add(new Map(modData, mapPackage));
+                                                       }
                                                }
                                        }
                                }
```

Reply:
Thanks. I forgot about empty folders. Fixed.